### PR TITLE
de-inline v8 helper functions

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* de-inline some V8 helper functions
* rename internal attribute "waitForCollector" to "flushColumnFamilies" (what it actually does). Does not change APIs
* use StaticStrings in a few more cases

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: -
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/ClusterEngine/ClusterRestWalHandler.cpp [hunk 1]
```diff
@@ -105,42 +106,33 @@ void ClusterRestWalHandler::flush() {
...
-    VPackSlice value = slice.get("waitForSync");
+    VPackSlice value = slice.get(StaticStrings::WaitForSyncString);
     if (value.isString()) {
-      waitForSync = (value.copyString() == "true");
+      waitForSync = (value.stringView() == "true");
     } else if (value.isBoolean()) {
       waitForSync = value.getBoolean();
     }
 
     value = slice.get("waitForCollector");
```
Comment (mpoeter) [inline arangod/ClusterEngine/ClusterRestWalHandler.cpp:120]: Also define a StaticString for "waitForCollector"?

File: arangod/ClusterEngine/ClusterRestWalHandler.cpp [hunk 2]
```diff
@@ -105,42 +106,33 @@ void ClusterRestWalHandler::flush() {
...
-    {
-      std::string const& v = _request->value("waitForCollector", found);
-      if (found) {
-        waitForCollector = (v == "1" || v == "true");
-      }
-    }
+    waitForSync =
+        _request->parsedValue(StaticStrings::WaitForSyncString, waitForSync);
+    flushColumnFamilies =
+        _request->parsedValue("waitForCollector", flushColumnFamilies);
```
Comment (mpoeter) [inline arangod/ClusterEngine/ClusterRestWalHandler.cpp:131]: StaticString?

File: arangod/RocksDBEngine/RocksDBRestWalHandler.cpp [hunk 3]
```diff
@@ -102,34 +102,35 @@ void RocksDBRestWalHandler::flush() {
...
-    VPackSlice value = slice.get("waitForSync");
+    VPackSlice value = slice.get(StaticStrings::WaitForSyncString);
     if (value.isString()) {
-      waitForSync = basics::StringUtils::boolean(value.copyString());
+      waitForSync = basics::StringUtils::boolean(value.stringView());
     } else if (value.isBoolean()) {
       waitForSync = value.getBoolean();
     }
 
     value = slice.get("waitForCollector");
```
Comment (mpoeter) [inline arangod/RocksDBEngine/RocksDBRestWalHandler.cpp:116]: StaticSstring?

File: arangod/RocksDBEngine/RocksDBRestWalHandler.cpp [hunk 4]
```diff
@@ -102,34 +102,35 @@ void RocksDBRestWalHandler::flush() {
...
     }
   } else {
     // no request body
-    waitForSync = _request->parsedValue("waitForSync", waitForSync);
-    waitForCollector =
-        _request->parsedValue("waitForCollector", waitForCollector);
+    waitForSync =
+        _request->parsedValue(StaticStrings::WaitForSyncString, waitForSync);
+    flushColumnFamilies =
+        _request->parsedValue("waitForCollector", flushColumnFamilies);
```
Comment (mpoeter) [inline arangod/RocksDBEngine/RocksDBRestWalHandler.cpp:127]: StaticString?

**Review by mpoeter — APPROVED** [review]

File: arangod/ClusterEngine/ClusterRestWalHandler.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/ClusterEngine/ClusterRestWalHandler.cpp:120]: I am a bit reluctant to define a static string value for an attribute that has the wrong name anyway.

Comment (mpoeter) [reply arangod/ClusterEngine/ClusterRestWalHandler.cpp:120]: I see pros and cons for both - so feel free to ignore this comment then.

**Pull request merged**